### PR TITLE
[#291] assume role

### PR DESCRIPTION
**Pull request by ErmiasG**

### CODE REVIEW

File: docs/integrations/role_mapping.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# IAM role mapping
+
+Using an EC2 instance profile enables your Hopsworks cluster to access AWS resources. This forces all Hopsworks users to
+share the instance profile role and the resource access policies attached to that role. To allow for per project access policies
+you could have your users use AWS credentials directly in their programs which is not recommended so you should instead use
+[Role chaining](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_terms-and-concepts.html#iam-term-role-chaining).
```
Comment (jimdowling) [inline docs/integrations/role_mapping.md:6]: (also known as [credential passthrough[(https://docs.databricks.com/security/credential-passthrough/iam-passthrough.html)).

File: docs/integrations/role_mapping.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+To use Role chaining, you need to first setup IAM roles in AWS:
+
+1. Create an instance profile role with policies that will allow it to assume all resource roles that we can assume from the Hopsworks
```
Comment (jimdowling) [inline docs/integrations/role_mapping.md:10]: This sentence is unwieldy.
Create an instance profile role that contains the different resource roles that we want to allow selected users to be able to assume in the Hopsworks cluster. In the example below, we define 4 different resource roles: `test-role`, `s3-role`, `dev-s3-role`, and `redshift` - and later we will define which users will be allowed assume which of these resources roles.

File: docs/integrations/role_mapping.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                    "arn:aws:iam::xxxxxxxxxxxx:role/dev-s3-role",
+                    "arn:aws:iam::xxxxxxxxxxxx:role/redshift"
+                ]
+            }
+        ]
+    }
+```
+Example policy for assuming four roles.
+
+2. Create the resource roles and edit trust relationship and add policy document that will allow the instance profile to assume this role.
```
Comment (jimdowling) [inline docs/integrations/role_mapping.md:34]: edit the trust relationship and add a policy document that will allow the instance profile to assume the resource roles.

File: docs/integrations/role_mapping.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            }
+        ]
+    }
+```
+Example policy document.
+
+3. Finally attach the instance profile to the master node of your Hopsworks AWS instance.
+
+
+Role chaining allows the instance profile to assume any role in the policy attached in step 1. To limit access
```
Comment (jimdowling) [inline docs/integrations/role_mapping.md:56]: to assume any of the 4 resource roles in the policy that was attached in step 1

Comment (jimdowling) [inline docs/integrations/role_mapping.md:56]: Typically, we will not want any user in Hopsworks to assume any of the resource roles. You can grant selected users the ability to assume any of the 4 resource roles from the admin page in hopsworks. In particular, we specify in which project(s) a given resource role can be used. Within a given project, we can further restrict who can assume the resource role by `mapping` the role to the group of users (`data owners` or `data scientists`).

File: docs/integrations/role_mapping.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+to roles we can create a per project mapping from the admin page in hopsworks.
+
+<p align="center">
+  <figure>
+    <img src="../../../assets/images/aws-role/goto-role-mapping.png" alt="Go to cloud role mapping.">
+    <figcaption>Go to cloud role mapping.</figcaption>
+  </figure>
+</p>
+
+By clicking the cloud role mapping icon in the admin page shown in the image above you can add mappings
```
Comment (jimdowling) [inline docs/integrations/role_mapping.md:66]: Can we change it from "cloud role mapping" ? It doesn't make any sense, IMO. 
For me, it's more "Granting roles to users" or "Grant assume role privileges to users" or "Mapping roles to projects/user-groups". 
If you want to emphasize that it is not AWS specific, Databricks use the term "credential passthrough".

File: docs/integrations/role_mapping.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+by entering the project name, which roles in that project can access the cloud role and the role ARN.
```
Comment (jimdowling) [inline docs/integrations/role_mapping.md:67]: which group of users in that project can assume the resource role and the role ARN.

What is the role ARN? It just appears here out of nowhere? cloud role hasn't been defined before this either.

File: docs/integrations/role_mapping.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+Optionally you can set a role mapping as default by marking the default checkbox. The default roles can be changed from the project setting by a Data owner in that project.
```
Comment (jimdowling) [inline docs/integrations/role_mapping.md:68]: Optionally, 
The default roles can only be changed by a Data owner who can do so in the project settings page.

File: docs/integrations/role_mapping.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+<p align="center">
+  <figure>
+    <img src="../../../assets/images/aws-role/role-mappings.png" alt="Add cloud role to project mapping.">
+    <figcaption>Add cloud role to project mapping.</figcaption>
```
Comment (jimdowling) [inline docs/integrations/role_mapping.md:73]: cloud role -> resource role ?
Discuss the cloud role term with others. I just find it unclear.

File: docs/integrations/role_mapping.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+Any member of a project can then go to the project settings page to see which roles they can assume.
+
+<p align="center">
+  <figure>
+    <img src="../../../assets/images/aws-role/project-cloud-roles.png" alt="Cloud roles mapped to project.">
+    <figcaption>Cloud roles mapped to project.</figcaption>
+  </figure>
+</p>
+
+For instructions on how to use the assume role API see [assuming a role](../assume_role/#assuming-a-role)
```
Comment (jimdowling) [inline docs/integrations/role_mapping.md:87]: missing a '.'

**Review by jimdowling — CHANGES_REQUESTED** [review]

File: docs/integrations/role_mapping.md (hunk 1 above)
Comment (ErmiasG) [reply docs/integrations/role_mapping.md:6]: Is it wise to reference databricks here?

Comment (jimdowling) [reply docs/integrations/role_mapping.md:6]: We are so small, i don't think it really matters (we really matter)!. But fine, leave it out.

**Review by jimdowling — APPROVED** [review]

**Comment by SirOibaf** [comment]
@ErmiasG - All the images in this PR are broken :( Also since the credential passthrough only applies to the storage connector, can we move the section under "Storage Connectors"?

**Pull request merged**